sonarts-core/tests/rules/noElementOverwrite/noElementOverwrite.lint.ts

```diff
@@ -0,0 +1,96 @@
+
+function test(cond: boolean): void {
+  let fruits = [];
+  fruits[1] = "banana";
+  fruits[1] = "apple";
+//^^^^^^ {{Verify this is the index that was intended; "1" was already set on line 4.}}
+
+  fruits = [];
+  fruits[1] = "banana";
+  fruits[1] = "apple";
+//^^^^^^ {{Verify this is the index that was intended; "1" was already set on line 9.}}
+
+  if (cond) {
+    fruits[1] = "potato"; // Compliant, because it's another block
+  }
+
+  fruits[2] = "orange";
+  fruits[2] = fruits[2] + ";"; // Compliant
+
+  for (let i = 0; i < 10; i++) {
+    fruits[i] = "melon";
+    fruits[i] = "pear";
```

Comment:
@saberduck let's add case when `i` is changed between 
also it would be nice to have something  like this:
```
fruits[i++] = 42;
fruits[i++] = 43;
```